### ISSUE TITLE
Feature: Add Native Bitcoin Wallet (linked to BTS private key)

### ISSUE THREAD

**Issue by btsfav** [bitshares/bitshares-ui#657]
So this would lay the foundation of atomic swaps between bitcoin and bitshares, and would open bitshares up to a greater community.

Every BitShares Account has a bitcoin address attached via its private key.

* extract the bitcoin address from the active private key
* add a Bitcoin wallet tab - with basic data send&receive
* add API access via https://api.greenaddress.it/index.html (or via witnesses later on)
* add an easy BTC <> OPEN.BTC/blocktrades exchange (optional)

Possible problems:
Password changes require key changes, which would erase your access to bitcoin. this needs to be addressed.

**Comment by kimchi-king** [comment]
This is a fantastic idea and is a much needed feature. I hope more people support this.

**Comment by CryptoKong** [comment]
Yeah i think this is a great idea, would love a Bitcoin QR code to send to as well.

**Comment by wmbutler** [comment]
Sounds like step one is to address the key changes. That discussion can happen here if need be. If we fail to address the key change, people will lose their BTC without even knowing it. Maybe we just check their address for a balance and disallow a key change if the balance > 0?

**Comment by btsfav** [comment]
@wmbutler balance >0 sounds like a very good idea to prevent this

**Comment by landry314** [comment]
But then how would someone change their password? Sure, transfer bitcoin out then back in to the new address... or a change in the next hardfork to take care of this automatically... I guess that is all we have.

**Comment by btsfav** [comment]
extra confirmation window then. there could be only dust sitting on an address

**Comment by startailcoon** [comment]
If this feature is possible, it would be a great addition.

**Comment by clockworkgr** [comment]
@sschiessl-bcp something to consider for Beet?

**Comment by sschiessl-bcp** [comment]
You are unearthing quite some gems here! Yes please. Give me 40hours in a day?
